daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SBuiltin.scala

```diff
@@ -1232,6 +1235,10 @@ object SBuiltin {
       machine.committers = extractParties(args.get(0))
       machine.commitLocation = optLocation
       machine.ctrl = CtrlValue.Unit
+      val inputContractIds = args.asScala.foldLeft(Set.empty[V.ContractId])(
```

Comment:
I don't understand why this code should have any effect, if the type signature in the commet of `SBSBeginCommit` is still up-to-date. Seeing that it extends `SBuiltin(2)`, there shouldn't be any contractIds in `args`. Am I missing something?
Or is it that the first argument of type `Party` can be a complex expression referring to contract IDs?

Reply:
You are right. I was too quick. 
Anyway, I am not sure I can implement the check of freshness for scenarios. 
I will probably do a first pass without scenario, and see in a second time if it can be added .

Reply:
contract id freshness is not does in case of scenarios.

Reply:
rewritten.